feat: Add TextLanguageClassifier 2.0

julian-risch opened the pull request:
### Related Issues

- First part of https://github.com/deepset-ai/haystack/issues/5677

### Proposed Changes:

Add TextLanguageClassifier component that takes list of string as input and routes to different output connections based on detected language. LangDetect is used for language detection.

### How did you test it?

Newly added unit tests and manually with the following code snippet:
```python
def test_pipe(self):
    document_store = MemoryDocumentStore()
    document_store.write_documents(documents=[Document(text="Emma lives in Berlin.")])
    p = Pipeline()
    p.add_component(instance=TextLanguageClassifier(), name="text_language_classifier")
    p.add_component(instance=MemoryBM25Retriever(document_store=document_store), name="retriever")
    p.connect("text_language_classifier.en", "retriever.query")
    result = p.run({"text_language_classifier": {"text": "Who lives in Berlin?"}})
    assert "Emma" in result["retriever"]["documents"][0].text
    result = p.run({"text_language_classifier": {"text": "Wer lebt in Berlin?"}})
    assert result == {}
```

### Notes for the reviewer

<!-- E.g. point out section where the reviewer  -->

### Checklist

- I have read the [contributors guidelines](https://github.com/deepset-ai/haystack/blob/main/CONTRIBUTING.md) and the [code of conduct](https://github.com/deepset-ai/haystack/blob/main/code_of_conduct.txt)
- I have updated the related issue with new insights and changes
- I added unit tests and updated the docstrings
- I've used one of the [conventional commit types](https://www.conventionalcommits.org/en/v1.0.0/) for my PR title: `fix:`, `feat:`, `build:`, `chore:`, `ci:`, `docs:`, `style:`, `refactor:`, `perf:`, `test:`.
- I documented my code
- I ran [pre-commit hooks](https://github.com/deepset-ai/haystack/blob/main/CONTRIBUTING.md#installation) and fixed any issue

anakin87 commented:
Hey, @julian-risch!

At first glance, the code looks fine.

Could you briefly explain the practical usage of this component?
- (a basic example would fit nicely in the class docstring)
- what are the use cases?
- how do you imagine it being used in a pipeline?
- how does this component relate to the batch/non-batch discussion? (#5818)

Feel free to skip these points. I'm just trying to get a better understanding :slightly_smiling_face:

julian-risch commented:
> Hey, @julian-risch!
> 
> At first glance, the code looks fine.
> 
> Could you briefly explain the practical usage of this component?
> 
> * (a basic example would fit nicely in the class docstring)
> * what are the use cases?
> * how do you imagine it being used in a pipeline?
> * how does this component relate to the batch/non-batch discussion? ([Interface batch and non-batch components #5818](https://github.com/deepset-ai/haystack/issues/5818))
> 
> Feel free to skip these points. I'm just trying to get a better understanding 🙂

@anakin87 Thank you for your feedback! I added an example to the class doc string and updated the PR description with an example that I used for testing. I also changed the input from List[str] to str. Makes more sense in connection with retriever in a pipeline used for query language classification. Decided to rename the input from "string" to "text". Could you please review again because I changed quite a few lines of code? Thank you!

anakin87 commented on haystack/preview/components/preprocessors/text_language_classifier.py:
```diff
@@ -0,0 +1,82 @@
+    For routing Documents based on their language use the related DocumentLanguageClassifier component.
+
+    Example usage in a retrieval pipeline that passes only English language queries to the retriever:
+    document_store = MemoryDocumentStore()
```
Nice example!.

I don't know if we need to prepend these examples with some special characters like ```

(@dfokina)

julian-risch replied:
Will do with ```python ... ``` 👍

anakin87 reviewed (approved):
Thanks for adding useful information: now everything is clear (and good)!

I just left a small comment about docstrings...
